### ISSUE TITLE
Allow passing a `returningColumns` array to insert, update, upsert and deletes shortcut functions

### ISSUE THREAD

**Issue by DanielPower** [jawj/zapatos#40]
The `select` shortcut function allows a select options object, which has various options such as `order`, `limit`, `columns`, etc. Most of these don't make sense for the `insert` shortcut function, since there's no need to filter or sort the result of insert. However it would be useful to be able to pass a `columns` array to the `insert` shortcut function.

Example:
`db.insert("user", { id: 5, name: "jawj", password: 58479 }, ["id", "name"]);`
`returns { id: 5, name: "jawj" }`;

This could of course be done manually by the user, but it would be a nice convenience.

**Comment by jawj** [comment]
Just to be clear, the implication here is that you want to insert all of the passed values but only return some of them, right?

I can see this is consistent with `select`, though if we did it, I think it would need to be very clearly signposted — e.g. an option named `returningColumns`. 

To be honest, I'm not sure how widely useful this would be. I'll have a think about it, and also see if anyone else want to chime in here.

It is otherwise just a one-line helper:

```typescript
const filterKeys = (obj, keys) => Object.fromEntries(Object.entries(obj).filter(([k, v]) => keys.includes(k)));

filterKeys({a: 1, b: 2, c: 3}, ['a', 'c']);  // => { a: 1, c: 3 }
```

**Comment by jawj** [comment]
OK, had a bit more of a think and a discussion with a colleague on this.

We're not wholly convinced by the example given, but we do think that efficiencies on the network and in JSON parsing are a good argument for adding a `returningColumns` option to `insert`, `update`, `upsert` and `deletes`. For example:

```typescript
const 
  thingColumn = getGiganticBlobThatDoesntNeedSendingRightBackToMe(),
  { id } = await db.insert('things', { thingColumn }, { returningColumns: ['id'] }).run(pool);
```

**Comment by DanielPower** [comment]
Sorry for the poor example, but yes, the goal is to reduce the amount of data retrieved from the db after an insert.

Given that `select` takes a property `columns` in its options object, I would think the property should be called columns for `insert`, `update`, etc. as well, for consistency. Or rename both to `returningColumns` for consistency.

As a note for anyone looking for a simple function to filter in the mean time without using a library like lodash, you can use this:

```
const object = { a: 1, b: 2, c: 3, d: 4 };
const simplifiedObject = (({ a, b }) => ({ a, b }))(object)
// simplifiedObject is { a: 1, b: 2 }
```

The syntax is a bit strange, but essentially you're passing the object into a function which destructures the desired properties, and creates a new object with only those properties.

**Comment by jawj** [comment]
I've now added and documented `returning` and `extras` options on the `insert`, `update`, `upsert` and `deletes` shortcuts, and released as 2.0.0.
